control-plane/connect-inject/controllers/pod/pod_controller_test.go

```diff
@@ -761,11 +763,1161 @@ func TestProxyConfigurationDelete(t *testing.T) {
+			//					Namespace: "ns1",
+			//					PeerName:  "peer1",
+			//				},
+			//				Name: "upstream1",
+			//			},
+			//			DestinationPort: "myPort",
+			//			Datacenter:      "",
+			//			ListenAddr: &pbmesh.Upstream_IpPort{
+			//				IpPort: &pbmesh.IPPortAddress{
+			//					Port: uint32(1234),
+			//				},
+			//			},
+			//		},
+			//	},
+			//},
+			consulNamespacesEnabled: true,
+			consulPartitionsEnabled: false,
+		},
+		{
+			name: "labeled annotated upstream with svc, ns, and partition",
+			pod: func() *corev1.Pod {
+				pod1 := createPod(podName, "1.2.3.4", "foo", true, true)
+				pod1.Annotations[constants.AnnotationMeshDestinations] = "myPort.port.upstream1.svc.ns1.ns.part1.ap:1234"
+				return pod1
+			},
+			expected: &pbmesh.Upstreams{
+				Workloads: &pbcatalog.WorkloadSelector{
+					Names: []string{podName},
+				},
+				Upstreams: []*pbmesh.Upstream{
+					{
+						DestinationRef: &pbresource.Reference{
+							Tenancy: &pbresource.Tenancy{
+								Partition: "part1",
+								Namespace: "ns1",
+								PeerName:  "",
+							},
+							Name: "upstream1",
+						},
+						DestinationPort: "myPort",
+						Datacenter:      "",
+						ListenAddr: &pbmesh.Upstream_IpPort{
+							IpPort: &pbmesh.IPPortAddress{
+								Port: uint32(1234),
+							},
+						},
+					},
+				},
+			},
+			consulNamespacesEnabled: true,
+			consulPartitionsEnabled: true,
+		},
+		{
+			name: "labeled annotated upstream with svc, ns, and dc",
+			pod: func() *corev1.Pod {
+				pod1 := createPod(podName, "1.2.3.4", "foo", true, true)
+				pod1.Annotations[constants.AnnotationMeshDestinations] = "myPort.port.upstream1.svc.ns1.ns.dc1.dc:1234"
+				return pod1
+			},
+			expErr: "upstream currently does not support datacenters: myPort.port.upstream1.svc.ns1.ns.dc1.dc:1234",
+			// TODO: uncomment this and remove expErr when datacenters is supported
+			//expected: &pbmesh.Upstreams{
+			//	Workloads: &pbcatalog.WorkloadSelector{
+			//		Names: []string{podName},
+			//	},
+			//	Upstreams: []*pbmesh.Upstream{
+			//		{
+			//			DestinationRef: &pbresource.Reference{
+			//				Tenancy: &pbresource.Tenancy{
+			//					Partition: "",
+			//					Namespace: "ns1",
+			//					PeerName:  "",
+			//				},
+			//				Name: "upstream1",
+			//			},
+			//			DestinationPort: "myPort",
+			//			Datacenter:      "dc1",
+			//			ListenAddr: &pbmesh.Upstream_IpPort{
+			//				IpPort: &pbmesh.IPPortAddress{
+			//					Port: uint32(1234),
+			//				},
+			//			},
+			//		},
+			//	},
+			//},
+			consulNamespacesEnabled: true,
+			consulPartitionsEnabled: false,
+		},
+		{
+			name: "labeled multiple annotated upstreams",
+			pod: func() *corev1.Pod {
+				pod1 := createPod(podName, "1.2.3.4", "foo", true, true)
+				pod1.Annotations[constants.AnnotationMeshDestinations] = "myPort.port.upstream1.svc.ns1.ns.dc1.dc:1234, myPort2.port.upstream2.svc:2234, myPort3.port.upstream3.svc.ns1.ns:3234, myPort4.port.upstream4.svc.ns1.ns.peer1.peer:4234"
+				return pod1
+			},
```

Comment:
I think we should have one functional case of multiple labeled upstream. Better yet, one that has both labeled and unlabeled would also be sufficient.

Reply:
oh yeah, I guess those did get removed with the ap/dcs being removed